timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/PaxosTimeLockServerIntegrationTest.java

```diff
@@ -79,11 +105,100 @@
     private final TimestampService timestampService = getTimestampService(CLIENT_1);
     private final TimestampManagementService timestampManagementService = getTimestampManagementService(CLIENT_1);
 
+    private static final LockRequest SLOW_REQUEST = LockRequest
+            .builder(ImmutableSortedMap.of(StringLockDescriptor.of("lock"), LockMode.WRITE))
+            .blockForAtMost(SimpleTimeDuration.of(100, TimeUnit.MILLISECONDS))
+            .build();
+
     @ClassRule
     public static final RuleChain ruleChain = RuleChain.outerRule(TEMPORARY_FOLDER)
             .around(TEMPORARY_CONFIG_HOLDER)
             .around(TIMELOCK_SERVER_HOLDER);
 
+    private TrustManager[] get_trust_mgr() {
+        TrustManager[] certs = new TrustManager[] {
+                new X509TrustManager() {
+                    public X509Certificate[ ] getAcceptedIssuers() { return null; }
+                    public void checkClientTrusted(X509Certificate[ ] certs, String t) { }
+                    public void checkServerTrusted(X509Certificate[ ] certs, String t) { }
+                }
+        };
+        return certs;
+    }
+
+    @Test
+    public void notExceedingThreadCountLimitsSucceeds() throws Exception {
+        List<RemoteLockService> lockService = ImmutableList.of(getLockService(CLIENT_1));
+
+        assertThat(lockAndUnlockAndCountExceptions(lockService, LOCAL_TC_LIMIT + GLOBAL_TC_LIMIT))
+                .isEqualTo(0);
+    }
+
+    @Test
+    public void multipleClientsCanShareGlobalThreads() throws Exception {
+        List<RemoteLockService> lockServiceList = ImmutableList.of(
+                getLockService(CLIENT_1), getLockService(CLIENT_2), getLockService(CLIENT_3));
+
+        assertThat(lockAndUnlockAndCountExceptions(lockServiceList, LOCAL_TC_LIMIT + GLOBAL_TC_LIMIT / 3))
+                .isEqualTo(0);
+    }
+
+    /*
+     * The following three tests are ignored as they flake too often.
+     */
+    @Ignore
+    @Test
+    public void exceedingThreadCountLimitsReturns429() throws Exception {
```

Comment:
Could this be managed with a unit test and a mocked lock service that acts like it has no available threads?